.werft/util/kubectl.ts

```diff
@@ -240,3 +241,62 @@ export function findFreeHostPorts(ranges: PortRange[], shellOpts: ExecOptions):
 export function waitForDeploymentToSucceed(name: string, namespace: string, type: string, shellOpts: ExecOptions) {
     exec(`kubectl rollout status ${type} ${name} -n ${namespace}`, shellOpts);
 }
+
+export async function waitUntilAllPodsAreReady(namespace: string, shellOpts: ExecOptions) {
+    interface Pod {
+        name: string
+        owner: string
+        phase: string
+    }
+    const werft = getGlobalWerftInstance();
```

Comment:
```suggestion
```

I've noticed that the trace result is broken: https://ui.honeycomb.io/gitpod/datasets/werft/trace/oH1d5CeEh9E?span=55953ea97f9ec6b5

Not sure about the reason, but could it be because we're not passing it as a parameter for this function and also for `waitForApiserver`? cc @mads-hartmann

Reply:
> could it be because we're not passing it as a parameter

not sure I understand what should be passed as a parameter here. Do you have an example? My understanding is that `werft` is a global static singleton and therefore does not need to be passed as a parameter.

Reply:
The `werft` instance is a singleton at the moment so passing it as parameter or fetching the global instance should be the same (I don't see any race conditions at a first glance at least). It is weird that your slices are orphaned and the `Ensuring VM is ready for deployment` phase doesn't show up in Honeycomb - so something is wrong in the instrumentation.